### PR TITLE
Fetch configuration from monika-hq

### PR DESCRIPTION
**Pull request by armedi**
implements issue #117 

- [x] Add a new key in the configuration file to specify the URL or IP address of Symon and the API Key to connect to Symon.
- [x] If Symon server is specified, Monika first performs handshake with Symon. See this [doc](https://nicnocquee.github.io/monika/monika-hq#handshake).
- [ ] After handshake, Monika performs the [report flow in interval](https://nicnocquee.github.io/monika/monika-hq#report-flow).
- [ ] On incident/recovery, Monika also performs the report flow.

Note:
- I added getConfig() function because later the configuration can be updated on the fly with data from monika-hq

### How to test?

- Setup a mock endpoint for monika-hq with [httptarget](https://github.com/hyperjumptech/httptarget)
  ```bash
  curl --location --request POST 'http://localhost:51423/api/paths' \
  --header 'Content-Type: application/json' \
  --data-raw '{
      "base_path": "monikahq/api/handshake",
      "delay_min_ms": 0,
      "delay_max_ms": 20,
      "return_code": 200,
      "return_headers": {
          "Content-Type": [
              "application/json"
          ]
      },
      "return_body": "{\"monika\":{\"id\":\"some_uuid\",\"ip_address\":\"this-instance-ip\"},\"data\":{\"probes\":[{\"requests\":[{\"url\":\"http://localhost:51423/delay/50/100\"}],\"incidentThreshold\":1,\"recoveryThreshold\":1,\"alerts\":[\"status-not-2xx\",\"response-time-greater-than-70-ms\"]}]}}"
  }'
  ```

- use this in monika configuration json file
  ```json
  {
    "monika-hq": {
      "url": "http://localhost:51423/monikahq",
      "key": "GENERATED_KEY"
    }
  }
  ```

- now when you run monika, it should use configuration from monika-hq

### CODE REVIEW

File: src/components/config/index.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    try {
+      const {
+        data: { probes, notifications },
+      } = await handshake(parsed)
+      if (probes) parsed.probes = probes
+      if (notifications) parsed.notifications = notifications
+    } catch (error) {
+      throw new Error(
+        'Please check your monika-hq server. It does not return valid configuration. Or remove "monika-hq" from configuration file'
+      )
```
Comment (nicnocquee) [inline src/components/config/index.ts:60]: Maybe better to just throw warning but use the old config file

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/components/config/index.ts (hunk 1 above)
Comment (armedi) [reply src/components/config/index.ts]: done

**Comment by armedi** [comment]
rebased to main branch

**Review by nicnocquee — APPROVED** [review]

**Comment by armedi** [comment]
rebased to main branch and resolved conflict

**Review by dennypradipta — APPROVED** [review]

**Pull request merged**